### PR TITLE
implement search addresses

### PR DESCRIPTION
**Pull request by AlexAzalor**

### CODE REVIEW

File: api/routes/location.py [hunk 1]
```diff
@@ -51,57 +51,110 @@ def get_all_locations(
...
-        cities_addresses_list.append(
-            s.CityAddressesOut(
-                city=s.City.model_validate(city),
-                addresses=[s.AddressOut.model_validate(address) for address in city_addresses],
+        settlements_data_out.append(
+            s.Settlement(
+                uuid=settlement.city_id,
+                location=f"{type}{settlement.name_ua}, район {rayon_name}, {region_name}",
             )
         )
```
Comment (Taraslysak) [inline api/routes/location.py:109]: This code looks very strange. What about English then?

File: api/routes/location.py [hunk 2]
```diff
@@ -51,57 +51,110 @@ def get_all_locations(
...
+    settlements: Sequence[m.Settlement] = db.scalars(sa.select(m.Settlement).where(*lookups).order_by(type_order)).all()
+
+    if not settlements:
+        log(log.INFO, "Settlements not found by query [%s]", query)
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Settlements not found")
+
+    rayons_ids = [settlement.district_id for settlement in settlements]
 
-    cities_ids = [city.city_id for city in cities]
+    rayons = db.scalars(sa.select(m.Rayon).where(m.Rayon.district_id.in_(rayons_ids))).all()
```
Comment (Taraslysak) [inline api/routes/location.py:85]: does settlement have relationship on rayron?

**Review by Taraslysak — COMMENTED** [review]
Wierd but ok for now

File: api/routes/location.py (hunk 2 above)
Comment (AlexAzalor) [reply api/routes/location.py:85]: no

**Pull request merged**